scipy/special/_support_alternative_backends.py

```diff
@@ -41,9 +41,9 @@ def get_array_special_func(f_name, xp, n_array_args):
     def __f(*args, _f=_f, _xp=xp, **kwargs):
         array_args = args[:n_array_args]
         other_args = args[n_array_args:]
-        array_args = [np.asarray(arg) for arg in array_args]
+        array_args = [xp_asarray(arg, xp=np) for arg in array_args]
         out = _f(*array_args, *other_args, **kwargs)
-        return _xp.asarray(out)
+        return xp_asarray(out, xp=xp)
```

Comment:
Outputs will typically be read-only after this step. This seems like a big disadvantage compared to `xp.asarray` using the array protocol. I don't think we want to return read-only results from special functions. Is there a general solution that is efficient (e.g. not just `copy` the output before returning it) or do we have to be very careful about using `from_dlpack` only when necessary?